components/hybrid_engine/src/engine_iterator.rs

```diff
@@ -97,3 +99,53 @@ where
         }
     }
 }
+
+pub struct HybridEngineIterMetricsCollector<EK, EC>
+where
+    EK: KvEngine,
+    EC: RangeCacheEngine,
+{
+    collector: Either<
+        <<EK::Snapshot as Iterable>::Iterator as MetricsExt>::Collector,
+        <<EC::Snapshot as Iterable>::Iterator as MetricsExt>::Collector,
+    >,
+}
+
+impl<EK, EC> IterMetricsCollector for HybridEngineIterMetricsCollector<EK, EC>
+where
+    EK: KvEngine,
+    EC: RangeCacheEngine,
+{
+    fn internal_delete_skipped_count(&self) -> usize {
+        match &self.collector {
+            Either::Left(c) => c.internal_delete_skipped_count(),
+            Either::Right(c) => c.internal_delete_skipped_count(),
+        }
+    }
+
+    fn internal_key_skipped_count(&self) -> usize {
+        match &self.collector {
+            Either::Left(c) => c.internal_key_skipped_count(),
+            Either::Right(c) => c.internal_key_skipped_count(),
+        }
+    }
+}
+
+impl<EK, EC> MetricsExt for HybridEngineIterator<EK, EC>
+where
+    EK: KvEngine,
+    EC: RangeCacheEngine,
+{
+    type Collector = HybridEngineIterMetricsCollector<EK, EC>;
+
+    fn metrics_collector(&self) -> Self::Collector {
```

Comment:
IMO, here we don't need to wrap a collector object `HybridEngineIterMetricsCollector` at all. 
Instead we can do this:
fn metrics_collector(&self) -> impl IterMetricsCollector

The current code is kind of manual implementation of Polymorphism.

Reply:
No. impl IterMetricsCollector is dynamic dispatch
@SpadeA-Tang